nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -183,4 +204,83 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
+     * @return {@code true} if comments by author were found, otherwise {@code false}
+     * @since 11.1
+     */
+    protected boolean hasCommentsByUser(CoreSession session, DocumentModel document, String author) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Handles autosubscription to {@value #COMMENT_ADDED_NOTIFICATION} following notification rules.
+     *
+     * @implSpec The default implementation autosubscribes top level document's author on first comment if
+     *           autosubscription is enabled. This relies on the implementation of the abstract hasComments method.<br/>
+     *           The default implementation autosubscribes comment author on his first comment on top level document if
+     *           autosubscription is enabled. This relies on the implementation of the abstract hasCommentsByUser method
+     * @param session the core session
+     * @param commentDocModel the comment
+     * @param document the document being commented
+     * @since 11.1
+     */
+    protected void handleNotificationAutoSubscriptions(CoreSession session, DocumentModel commentDocModel,
+            DocumentModel document) {
+        if (Framework.getService(ConfigurationService.class).isBooleanFalse(AUTOSUBSCRIBE_CONFIG_KEY)) {
+            log.trace("autosubscription to new comments is disabled");
+            return;
+        }
+
+        NuxeoPrincipal topLevelDocumentAuthor = getAuthor(document);
+        if (!hasComments(session, document)) {
+            // Document author is subscribed on first comment by anybody
+            subscribeCommentAddedNotification(document, topLevelDocumentAuthor);
+
+        }
+
+        NuxeoPrincipal commentAuthor = getAuthor(commentDocModel);
+        if (topLevelDocumentAuthor.getName().equals(commentAuthor.getName())) {
+            // Document author is comment author. He doesn't need to be resubscribed
+            return;
+        }
+
+        if (!hasCommentsByUser(session, document, commentAuthor.getName())) {
+            // Comment author is writing his first comment on the document
+            subscribeCommentAddedNotification(document, commentAuthor);
+        }
+    }
+
+    /**
+     * Subscribes a user to {@value #COMMENT_ADDED_NOTIFICATION} notifications on the document.
+     *
+     * @param document the document being commented
+     * @param user the user to subscribe to comment notifications
+     * @since 11.1
+     */
+    protected void subscribeCommentAddedNotification(DocumentModel document, NuxeoPrincipal user) {
+        String subscriber = NotificationConstants.USER_PREFIX + user.getName();
+        NotificationManager notificationManager = Framework.getService(NotificationManager.class);
+        notificationManager.addSubscription(subscriber, COMMENT_ADDED_NOTIFICATION, document, false, user,
+                COMMENT_ADDED_NOTIFICATION);
```

Comment:
sorry, i have an other interrogation about this auto-subscribe: actually if you use the platform (i made my test on my local 10.10) when a user clique on the  `Notify Me` button the backend code that will be called will be 

`org.nuxeo.ecm.platform.ec.notification.service.NotificationService#addSubscriptions((String username, DocumentModel doc, Boolean sendConfirmationEmail,
            NuxeoPrincipal principal))`
through the Subscription operation. we don't give the name of the notification and then the method that will be called is `doc.getAdapter(SubscriptionAdapter.class).addSubscriptionsToAll(username);` which will add the notification list below:

`"Modification", "CommentPublication", "Approbation review started", "CommentUpdated", "Creation", "CommentAdded", "Workflow Change"`

my interrogations:

- In your code yo give the `CommentAdded` when you auto-subscribe if the treatment is not the same then the web-ui,  what about `CommentUpdated`, an user can update a given comment under some conditions (admin ...) 
- do we need to send a mail to notify that the user is now auto-subscribed: `.addSubscription(subscriber, COMMENT_ADDED_NOTIFICATION, document, false, user,
                COMMENT_ADDED_NOTIFICATION);`
`false` means no mail will be sent when we subscribe 
-

Reply:
Asked Alain about it, we will subscribe to all to be consistent with the notify me button !
About second point, we are going to use SubscriptionAdapter#addSubscriptionsToAll which only takes a username and doesn't notify